lib/services/services.c

```diff
@@ -406,13 +406,20 @@ services__create_resource_action(const char *name, const char *standard,
             GHashTableIter iter;
             char *key = NULL;
             char *value = NULL;
-            int index = 1;
-            static int args_size = sizeof(op->opaque->args) / sizeof(char *);
+            int index = 1; // 0 is already set to executable name
 
             g_hash_table_iter_init(&iter, op->params);
 
-            while (g_hash_table_iter_next(&iter, (gpointer *) & key, (gpointer *) & value) &&
-                   index <= args_size - 3) {
+            while (g_hash_table_iter_next(&iter, (gpointer *) & key, (gpointer *) & value)) {
+
+                if (index > (PCMK__NELEM(op->opaque->args) - 2)) {
```

Comment:
We know the hash table size - right? So we could catch that ahead.

Reply:
We skip meta-attributes, so we do have to iterate

Reply:
Shouldn't be a problem in real life anyway, a nagios plugin with 128 command-line options would be an unusual sight. :) It was fun to create one for testing though

Reply:
Thus I thought we could as well be generous about the meta-attributes just to stay on the safe side.